packages/react-core/src/hooks/useUserAttributes/action.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  confirmUserAttribute,
+  deleteUserAttributes,
+  fetchUserAttributes,
+  sendUserAttributeVerificationCode,
+  updateUserAttributes,
+  UserAttributeKey,
+} from '@aws-amplify/auth';
+
+import {
+  ConfirmInput,
+  DeleteInput,
+  FetchInput,
+  HandleAttributeActionInput,
+  HandleAttributeActionOutput,
+  SendCodeInput,
+  UpdateInput,
+  VerifiableAttribute,
+} from './types';
+import { UpdateUserAttributesOutput } from 'aws-amplify/auth';
+
+const getVerifiableAttribute = (
+  attributes: UpdateUserAttributesOutput,
+  name: 'email' | 'phone_number'
+): VerifiableAttribute | undefined => {
+  const attribute = attributes[name];
+  if (
+    attribute &&
+    attribute.nextStep &&
+    attribute.nextStep.updateAttributeStep === 'CONFIRM_ATTRIBUTE_WITH_CODE' &&
+    attribute.nextStep.codeDeliveryDetails
+  ) {
+    const { deliveryMedium, destination } =
+      attribute.nextStep.codeDeliveryDetails;
+    return {
+      name,
+      codeDeliveryDetails: {
+        medium: deliveryMedium as string,
+        destination: destination as string,
+      },
+    };
+  } else {
+    return undefined; // Extraneous
```

Comment:
Can this be removed if it's extraneous?

Reply:
Thanks for pointing that out, meant to remove it before and it's gone now

Reply:
Also cleaned up the conditional logic above it